test/sql/copy/csv/test_date_sniffer.test

```diff
@@ -0,0 +1,50 @@
+# name: test/sql/copy/csv/test_date_sniffer.test
+# description: Test sniffer bug from date detection is issue https://github.com/duckdb/duckdb/issues/13121
+# group: [csv]
+
+statement ok
+PRAGMA enable_verification
+
+query III
+FROM 'data/csv/bad_date.csv'
+----
+2016-10-03	vst_OSBS_2016	41101
```

Comment:
Turning `41_10_1` into a number is likely also not intentional. Unrelated to this specific issue but maybe we should move the underscore separation behind `strict` casts as well?

Reply:
Ah, I had done this already in the first commit, but removed it since I the cast seemed to be expected anyway.

I'll add it back.